infra/event-sync/src/test/java/com/simprints/infra/eventsync/sync/down/workers/CommCareEventSyncDownloaderWorkerTest.kt

```diff
@@ -0,0 +1,216 @@
+
+    @MockK
+    lateinit var commCareSyncTask: CommCareEventSyncTask
+
+    @MockK
+    lateinit var eventDownSyncScopeRepository: EventDownSyncScopeRepository
+
+    @MockK
+    lateinit var syncCache: EventSyncCache
+
+    @MockK
+    lateinit var eventRepository: EventRepository
+
+    @MockK
+    lateinit var eventScope: EventScope
+
+    @MockK
+    lateinit var configRepository: ConfigRepository
+
+    @MockK
+    lateinit var migrationFlagsStore: RealmToRoomMigrationFlagsStore
+
+    private lateinit var eventDownSyncDownloaderWorker: CommCareEventSyncDownloaderWorker
+
+    @Before
+    fun setUp() {
+        MockKAnnotations.init(this, relaxed = true)
+
+        eventDownSyncDownloaderWorker = CommCareEventSyncDownloaderWorker(
+            mockk(relaxed = true),
+            mockk(relaxed = true) {
+                every { inputData } returns workDataOf(
+                    INPUT_DOWN_SYNC_OPS to JsonHelper.toJson(projectDownSyncScope.operations.first()),
+                    INPUT_EVENT_DOWN_SYNC_SCOPE_ID to "eventScopeId",
+                )
+            },
+            eventDownSyncScopeRepository,
+            syncCache,
+            JsonHelper,
+            eventRepository,
+            configRepository,
+            testCoroutineRule.testCoroutineDispatcher,
+            commCareSyncTask,
+            migrationFlagsStore,
+        )
+    }
+
+    @Test
+    fun `worker when migration is in progress should retry and not set down sync in progress`() = runTest {
+        coEvery { migrationFlagsStore.isMigrationInProgress() } returns true
+
+        val result = eventDownSyncDownloaderWorker.doWork()
+
+        assertThat(result).isEqualTo(ListenableWorker.Result.retry())
+        coVerify(exactly = 0) { migrationFlagsStore.setDownSyncInProgress(true) }
+        coVerify(exactly = 0) { migrationFlagsStore.setDownSyncInProgress(false) }
+    }
+
+    @Test
+    fun `worker when migration is not in progress should set down sync in progress to true then false`() = runTest {
+        coEvery { migrationFlagsStore.isMigrationInProgress() } returns false
+
+        eventDownSyncDownloaderWorker.doWork()
+
+        coVerify(exactly = 1) { migrationFlagsStore.setDownSyncInProgress(true) }
+        coVerify(exactly = 1) { migrationFlagsStore.setDownSyncInProgress(false) }
+    }
+
+    @Test
+    fun `worker when migration is not in progress and exception occurs should set down sync in progress to true then false`() = runTest {
+        coEvery { migrationFlagsStore.isMigrationInProgress() } returns false
+        coEvery { commCareSyncTask.downSync(any(), any(), any(), any()) } throws RuntimeException("Test exception")
+
+        eventDownSyncDownloaderWorker.doWork()
+
+        coVerify(exactly = 1) { migrationFlagsStore.setDownSyncInProgress(true) }
+        coVerify(exactly = 1) { migrationFlagsStore.setDownSyncInProgress(false) }
+    }
+
+    @Test
+    fun worker_noEventScope_shouldFail() = runTest {
```

Comment:
Naming should be consistent at least within the file, and I would suggest sticking to ticks as it is a more flexible option.